packages/developer-portal/src/components/pages/app-detail/app-content-v8.tsx

```diff
@@ -0,0 +1,217 @@
+
+export type AppContentProps = {
+  appDetailState: AppDetailState
+}
+
+export type CustomUninstallCell = React.FC<{ onClick: () => void }>
+
+interface HandleUninstallSuccessParams {
+  handleAfterClose: any
+  setUninstallApp: React.Dispatch<React.SetStateAction<InstallationModel | undefined>>
+  developerId: string
+  appId: string
+  dispatch: Dispatch
+}
+
+export const handleUninstallSuccess =
+  ({ handleAfterClose, setUninstallApp, developerId, appId, dispatch }: HandleUninstallSuccessParams) =>
+  () => {
+    handleAfterClose({ setUninstallApp })
+    dispatch(
+      fetchInstallationsList({
+        appId: [appId],
+        pageNumber: 1,
+        pageSize: GET_ALL_PAGE_SIZE,
+        isInstalled: true,
+        developerId: [developerId],
+      }),
+    )
+  }
+
+export const handleOpenAppPreview = (appId: string, appDetailState: AppDetailModel) => () => {
+  const url = routes.APP_PREVIEW.replace(':appId', appId)
+  localStorage.setItem('developer-preview-app', JSON.stringify(appDetailState))
+  window.open(url, '_blank')
+}
+
+export const CustomUninstallCell: React.FC<{ onClick: () => void }> = ({ onClick }) => (
+  <a onClick={onClick}>Uninstall</a>
+)
+
+export const generateInstallationTableColumns =
+  (onUninstall: (app: InstallationModel) => () => void, CustomUninstallCell?: CustomUninstallCell) => () => {
+    const UninstallCell = ({ row }) => {
+      if (CustomUninstallCell) {
+        return <CustomUninstallCell onClick={onUninstall(row.original)} />
+      }
+
+      return (
+        <Button type="button" intent="primary" onClick={onUninstall(row.original)}>
+          Uninstall
+        </Button>
+      )
+    }
+
+    return [
+      {
+        Header: 'Client',
+        accessor: 'client',
+      },
+      {
+        Header: 'Company Name',
+        accessor: 'customerName',
+      },
+      {
+        Header: 'Company Address',
+        accessor: ({ customerAddress = {} }: { customerAddress: InstallationModel['customerAddress'] }) => {
+          const {
+            buildingName = '',
+            buildingNumber = '',
+            line1 = '',
+            line2 = '',
+            line3 = '',
+            line4 = '',
+            postcode = '',
+            countryId = '',
+          } = customerAddress
+
+          return `${buildingName} ${buildingNumber} ${line1} ${line2} ${line3} ${line4} ${postcode} ${countryId}`
+        },
+      },
+      {
+        Header: 'Date Installed',
+        accessor: (d) => dayjs(d.created).format(DATE_TIME_FORMAT.DATE_FORMAT),
+      },
+      {
+        Header: 'Installed By',
+        accessor: 'installedBy',
+      },
+      {
+        Header: 'Uninstall',
+        Cell: UninstallCell,
+      },
+    ]
+  }
+
+const AppContent: React.FC<AppContentProps> = ({ appDetailState }) => {
+  const appDetailData = appDetailState.data || {}
+
+  return (
+    <>
+      <Grid>
+        <ColSplit>
+          <InputWrap>
```

Comment:
So I don't think you need input wrap - that's really for, well, inputs!

Reply:
I'm not too worried TBH, it's a placeholder file I appreciate

Reply:
Is there anything similar to this I can use instead?